### PR TITLE
Add export of particle dictionaries as pickles

### PR DESCRIPTION
**Pull request by awirb**
exports a pickle for each particle dictionary (for each dtSave) to Outputs/com1DFAPy/particles 

* particle dicionaries can be read with readPartFromPickle which returns a list of particle dictionaries and a list of corresponding time steps

### CODE REVIEW

File: avaframe/com1DFAPy/com1DFA.py [hunk 1]
```diff
@@ -1097,6 +1099,43 @@ def removePart(particles, mask, nRemove):
...
+            list of dictionaries or single dictionary
+        outDir: str
+            path to output directory
+    """
+    if len(dictList) > 1:
+        for dict in dictList:
+            pickle.dump(dict, open(os.path.join(outDir, "particles%f.p" % dict['t']), "wb"))
+    else:
+        pickle.dump(dictList, open(os.path.join(outDir, "particles.p"), "wb"))
+
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1117]: If there's only one dict, there's also a time information, why not include this?

File: avaframe/com1DFAPy/com1DFA.py [hunk 2]
```diff
@@ -1097,6 +1099,43 @@ def removePart(particles, mask, nRemove):
...
+
+def readPartFromPickle(inDir, flagAvaDir=False):
+    """ Read pickles within a directory and return List of dicionaries read from pickle """
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1120]: Please document the flagAvaDir override (explain what it does..)

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFAPy/com1DFA.py (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1117]: included

File: avaframe/com1DFAPy/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1120]: done

File: avaframe/com1DFAPy/com1DFA.py [hunk 3]
```diff
@@ -1097,6 +1099,52 @@ def removePart(particles, mask, nRemove):
...
+def readPartFromPickle(inDir, flagAvaDir=False):
+    """ Read pickles within a directory and return List of dicionaries read from pickle
+
+        Parameters
+        -----------
+        inDir: str
+            path to input directory
+        flagAvaDir: bool
+            if True inDir corresponds to an avalanche directory and pickles are
+            saved to avaDir/Outputs/com1DFAPy/particles
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:1128]:
```suggestion
            read from avaDir/Outputs/com1DFAPy/particles 
```

**Review by fso42 — APPROVED** [review]
Small typo

File: avaframe/com1DFAPy/com1DFA.py (hunk 3 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1128]: ah yes, thanks!

**Pull request merged**